### ISSUE TITLE
Timestamp comparison of config files logs a wrong message

### ISSUE THREAD

**Issue by baurmatt** [Icinga/icinga2#5484]
@dnsmichi ask me to open up a issue about this. After upgrading to 2.7.0 today, i found strange messages in the log file of our satellites:

```
[2017-08-09 11:08:16 +0200] information/ApiListener: Cannot apply configuration file update for path '/var/lib/icinga2/api/zones/globals.d'. Current timestamp '2017-08-09 10:35:13 +0200' (1502267713.652104) is more recent than received timestamp '2017-08-09 10:35:13 +0200' (1502267713.652104).
```

The timestamps are equal but for some $reasons Icinga2 thinks they're different.

## Your Environment
* Version used (`icinga2 --version`):
```
root@monitoring-satellite02:/# dpkg -l | grep icinga
ii  icinga2                              2.7.0-1.xenial                             amd64        host and network monitoring system
ii  icinga2-bin                          2.7.0-1.xenial                             amd64        host and network monitoring system - daemon
ii  icinga2-common                       2.7.0-1.xenial                             all          host and network monitoring system - common files
ii  libicinga2                           2.7.0-1.xenial                             amd64        host and network monitoring system - internal libraries
```
* Operating System and version: Ubuntu 16.04.2
* Enabled features (`icinga2 feature list`):
```
root@monitoring-satellite02:/# icinga2 feature list
Disabled features:
Enabled features: api checker mainlog
```

**Comment by dnsmichi** [comment]
@Crunsher seems the recent log addition for config updates logs an error if the timestamp is equal. Should only do for less equal, but requires further testing.

**Comment by Crunsher** [comment]
Damn.
Introduced by #5211

**Comment by dnsmichi** [comment]
I would still ignore configs which have an equal timestamp. I think users are irritated because of the new configuration messages which previously has been a no-op.

This isn't really a bug in the config sync, it is just wrong logging.

**Comment by Crunsher** [comment]
My worry is that this affects overrides from global zones, but I really need to investigate this first.

**Comment by dnsmichi** [comment]
This is another issue you've been working on at a customer in the past month, right?

**Comment by Crunsher** [comment]
This is a separate issue, while this was introduced as a possible fix for one, it's not the same.

**Comment by vkafed** [comment]
have the same bug... :-(

**Comment by tobiasvdk** [comment]
I think it's just the new log messages which irritates since the comparison was not changed recently.
```diff
$ git log -L101,101:lib/remote/apilistener-filesync.cpp
commit 7c70d51b2713b9a47527ee7435adc2129d5cb006
Author: Jean Flach <jean-marcel.flach@icinga.com>
Date:   Fri Apr 28 16:51:23 2017 +0200

    Add a few extra log messages
    
    fixes #5212

diff --git a/lib/remote/apilistener-filesync.cpp b/lib/remote/apilistener-filesync.cpp
--- a/lib/remote/apilistener-filesync.cpp
+++ b/lib/remote/apilistener-filesync.cpp
@@ -101,1 +101,1 @@
-       if (oldTimestamp >= newTimestamp)
+       if (oldTimestamp >= newTimestamp) {

commit d412a9055d2e55b2291bdcd31643ed5e2d955a8f
Author: Michael Friedrich <michael.friedrich@netways.de>
Date:   Mon Jan 25 15:21:25 2016 +0100

    Fix cluster config timestamp again
    
    refs #11014

diff --git a/lib/remote/apilistener-filesync.cpp b/lib/remote/apilistener-filesync.cpp
--- a/lib/remote/apilistener-filesync.cpp
+++ b/lib/remote/apilistener-filesync.cpp
@@ -68,1 +76,1 @@
-       if (oldTimestamp <= newTimestamp)
+       if (oldTimestamp >= newTimestamp)

commit a17419e8c23af7dc236bd93f9889aada21a25114
Author: Michael Friedrich <michael.friedrich@netways.de>
Date:   Mon Jan 25 14:25:37 2016 +0100

    Fix cluster config sync timestamp
    
    refs #11014

diff --git a/lib/remote/apilistener-filesync.cpp b/lib/remote/apilistener-filesync.cpp
--- a/lib/remote/apilistener-filesync.cpp
+++ b/lib/remote/apilistener-filesync.cpp
@@ -65,4 +68,1 @@
-               /* skip update if our config is newer */
-               if (oldTS <= newTS)
-                       return false;
-       }
+       if (oldTimestamp <= newTimestamp)

commit 14cb049b6a93485342c56f3f65b68ce751e5d0e4
Author: Gunnar Beutner <gunnar.beutner@netways.de>
Date:   Thu May 15 14:27:50 2014 +0200

    Implement timestamps for zone configs.
    
    Fixes #6191

diff --git a/lib/remote/apilistener-sync.cpp b/lib/remote/apilistener-sync.cpp
--- a/lib/remote/apilistener-sync.cpp
+++ b/lib/remote/apilistener-sync.cpp
@@ -60,0 +65,4 @@
+               /* skip update if our config is newer */
+               if (oldTS <= newTS)
+                       return false;
+       }
```

Nevertheless, do same timestamps mean, that the config is the same or is it necessary to create a checksum?

**Comment by dnsmichi** [comment]
Edited the subject, it really only is a matter of the new log message here.
